tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,370 @@
+
+                // If the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                oracle1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Directory.CreateTempSubdirectory().FullName;
+
+                if (!OperatingSystem.IsWindows())
+                {
+                    // Change permissions for non-root accounts (container user account)
+                    File.SetUnixFileMode(bindMountPath, MountFilePermissions);
+                }
+
+                oracle1.WithDataBindMount(bindMountPath);
+            }
+
+            using (var app = builder1.Build())
+            {
+                await app.StartAsync();
+
+                await app.WaitForText("Completed: ALTER DATABASE OPEN", cancellationToken: cts.Token);
+
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+                    {
+                        [$"ConnectionStrings:{db1.Resource.Name}"] = await db1.Resource.ConnectionStringExpression.GetValueAsync(default)
+                    });
+
+                    hb.AddOracleDatabaseDbContext<TestDbContext>(db1.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        using var dbContext = host.Services.GetRequiredService<TestDbContext>();
+
+                        // Wait until the database is available
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            return await dbContext.Database.CanConnectAsync(cts.Token);
+                        }, cts.Token);
+
+                        // Create tables
+                        await dbContext.Database.EnsureCreatedAsync(cts.Token);
+
+                        // Seed database
+                        dbContext.Cars.Add(new TestDbContext.Car { Brand = "BatMobile" });
+                        await dbContext.SaveChangesAsync(cts.Token);
+
+                        // Stops the container and wait until it's not accessible anymore before creating a new one
+                        // using the same volume.
+
+                        await app.StopAsync();
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+                            return !await dbContext.Database.CanConnectAsync(token);
+                        }, cts.Token);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
```

Comment:
We stopped it abvoe on line 162. Why do we need to stop it again? Repeated elsewhere.

Reply:
In case there is an exception before StopAsync is called we keep it in the finally. The outer finally block will then try to delete the volume.